newsrevenuehub/rev-engine
# [#9729] call verifyapplepaydomain on revenueprogram creation

#### What's this PR do?
Performs the necessary Stripe API call for Apple Pay domain verification whenever a new RevenueProgram is created. Does not handle tear-down.

#### How should this be manually tested?
This must be tested with a live key. Stripe does not allow us to call the ApplePayDomain.create endpoint with a test key. But in prod, just create a new revenue program and verify that it has a "apple_pay_domain_verification_date" that is not blank in Django admin. One should probably also go a step further and verify that apple pay appears on a donation page for that rev program-- but again, only testable live.

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?
We've added a DOMAIN_APEX config variable that should be "revengine.org" for prod and "revengine-test.org" for staging. These should already be set in Heroku.
